testing/trino-tests/src/test/resources/sql/presto/tpcds/hive/partitioned/q01.plan.txt

```diff
@@ -11,10 +11,10 @@ local exchange (GATHER, SINGLE, [])
                                     local exchange (GATHER, SINGLE, [])
                                         remote exchange (REPARTITION, HASH, ["sr_customer_sk", "sr_store_sk"])
                                             partial aggregation over (sr_customer_sk, sr_store_sk)
-                                                join (INNER, REPLICATED):
+                                                join (INNER, PARTITIONED):
                                                     scan store_returns
                                                     local exchange (GATHER, SINGLE, [])
-                                                        remote exchange (REPLICATE, BROADCAST, [])
+                                                        remote exchange (REPARTITION, local:HivePartitioningHandle{buckets=1, hiveTypes=[], partitions=2004}, ["d_date_sk"])
```

Comment:
A lot of TPC plans have changed from
```
join (INNER, REPLICATED):
    scan partitioned_fact_table
    local exchange (GATHER, SINGLE, [])
         remote exchange (REPLICATE, BROADCAST, [])
             scan date_dim
```             
to
```
join (INNER, PARTITIONED):
    scan partitioned_fact_table
    local exchange (GATHER, SINGLE, [])
         remote exchange (REPARTITION, local:HivePartitioningHandle{buckets=1, hiveTypes=[], partitions=2004}, ["d_date_sk"])
             scan date_dim
```
That might be a worse plan because usually we want to broadcast small build side rather than repartition them.

Reply:
What would you suggest here? The same thing would happen in this case if the probe table was bucketed, so this seems to me like a more general planner change. I was hoping to keep this PR in the Hive connector rather than the planner as much as possible. Maybe we can profile and see how urgent this problem is. Another option is disabling this feature by default pending planner improvements.

Reply:
Following up on this...anything I can do? I have rebased the PR against current master.

Reply:
Ideally we want features to ship as enabled by default so that all users benefit from it and it gets sufficient validation in  a variety of real world workloads.
If other maintainers agree, we could choose to keep this disabled by default.
However, this still requires proving performance benefits of this optimization and having green CI with this feature turned on so that we know there aren't surprises waiting for us when we try to enable it by default later.
E.g. It looks like right now there are product test failures with transactional tables, this needs to be resolved.